systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java

```diff
@@ -174,6 +183,162 @@ void testJSONFormatLogging() {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilOff);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilOff);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+        );
+
+        LOGGER.info("Waiting {} ms for INFO log will disappear", RECONCILIATION_INTERVAL * 2);
+        //wait some time if TO and UO will log something
+        Thread.sleep(RECONCILIATION_INTERVAL * 2);
+
+        LOGGER.info("Asserting if log is without records");
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "topic-operator", "30s"), is(emptyString()));
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "user-operator", "30s"), is(emptyString()));
+
+        LOGGER.info("Changing rootLogger level to INFO with inline logging");
+        InlineLogging ilInfo = new InlineLogging();
+        ilInfo.setLoggers(Collections.singletonMap("rootLogger.level", "INFO"));
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilInfo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilInfo);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=INFO")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=INFO")
+        );
+
+        LOGGER.info("Waiting {} ms for INFO log will appear", RECONCILIATION_INTERVAL * 2);
+        //wait some time if TO and UO will log something
+        Thread.sleep(RECONCILIATION_INTERVAL * 2);
+
+        LOGGER.info("Asserting if log will contain some records");
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "user-operator", "5m"), is(not(emptyString())));
```

Comment:
Why is this waiting for 5 minutes while for no logs only 30 seconds? If 30 seconds are good there, shouldn't they be good also here and vice versa?

Reply:
I did this change -> if I remember what `--since` takes last x seconds/minutes/hours until the moment you execute ... so I thought that will cause some race conditions when we set only 30s to get log and the EO will not log anything in that time. I can try that and see, if I'm correct.

Reply:
Right. But, when switching the logging OFF, you approach it that 30s without any log message is enough to take it as configured. But when enabling the INFO level, you seem to suggest that 5 minutes are needed to ensure there is some log message. If you need 5 minutes to be sure you got some message, you also need 5 minutes to ensure you got no message, or?

Reply:
Yeah, you are right, I gonna edit this :) thanks Jakub.